src/number_format.js

```diff
@@ -55,6 +55,23 @@ const propTypes = {
     PropTypes.func, // for legacy refs
     PropTypes.shape({ current: PropTypes.any }),
   ]),
+  customNumerals: (props, propName, componentName) => {
+    if (!props[propName]) {
+      return;
+    }
+    const arrayPropLength = props[propName].length;
+    const itemsAreString = props[propName].every((item) => typeof item === 'string');
```

Comment:
We might also want to look if it is a single character string. Multi character string (like Roman numerals) will not work.
```suggestion
    const hasSingleCharString = props[propName].every((item) => typeof item === 'string' && item.length === 1);
```